frame/support/src/traits.rs

```diff
@@ -934,36 +934,68 @@ impl WithdrawReasons {
 	}
 }
 
-/// Trait for type that can handle incremental changes to a set of account IDs.
-pub trait ChangeMembers<AccountId: Clone + Ord> {
+/// Trait for types that can handle incremental changes to a set of account IDs.
+///
+/// Allows member changes according to the "verify first, write last" maxime by
+/// providing the `ensure_can_change_members` function for checking as well as the
+/// actual writing logic.
+pub trait ChangeMembers<AccountId: Clone + Ord, MembersCount = usize> {
+
+	/// Returns whether changing the members will execute without errors.
+	///
+	/// Defaults to a noop. Implement this if `change_members_sorted` could fail.
+	///
+	/// If this returns `Err`, calls to `change_members`, `change_members_sorted` and 
+	/// `set_members_sorted` are not guaranteed to have the desired effect.
+	fn ensure_can_change_members(_new: &[AccountId], _old: &[AccountId]) -> DispatchResult {
```

Comment:
Why a result with `ensure_can_*` rather than a bool with `can_*`?

Reply:
I figured it would be more useful to get an indication of the error. The use case I imagine is similar to #5802 where we want to propagate the ensure error.
It is also more generic/open to (similar) extension in the future. There might be a different reason why the change members could fail.

Reply:
that seems reasonable. I dunno about api design, but I have seen most other traits use `bool` rather than `result`.

Dispatch result in particular is weird because you cannot really know what kinds of errors to expect to get in return from a random pallet implementing this trait. (since they are all custom)

I would confirm if one choice is better than the other from someone with more sense on API design